include/bluetooth/conn.h

```diff
@@ -297,6 +297,29 @@ int bt_conn_security(struct bt_conn *conn, bt_security_t sec);
  */
 u8_t bt_conn_enc_key_size(struct bt_conn *conn);
 
+enum {
+	/* Authentication failed. */
+	BT_CONN_AUTH_ERR_AUTHENTICATION_FAIL,
+
+	/* Required input was not available.  */
+	BT_CONN_AUTH_ERR_AUTHENTICATION_MISSING,
```

Comment:
Id rename this BT_CONN_AUTH_ERR_KEY_MISSING or perhaps BT_CONN_AUTH_ERR_INVALID, however I recall that some stacks do attempt to generate new keys when such error is generated, and in case of success replace the keys.

Reply:
In the LE case, this would be returned when OOB data is not available. Maybe KEY_OR_DATA_MISSING?

Reply:
Im usually lean towards sorter names whenever possible as it usually produces code with least of lines and in particular less multi-line identation to deal with.